sdk/internal/mock/mock.go

```diff
@@ -58,6 +60,27 @@ func NewTLSServer() (*Server, func()) {
 	return s, func() { s.srv.Close() }
 }
 
+// NewTLSHTTP2Server creates a new Server object with HTTP2 configuration.
+// The returned close func must be called when the server is no longer needed.
+// It will return nil for both the server and close func if it encountered an error
+// when configuring HTTP2 for the new TLS server.
+func NewTLSHTTP2Server() (*Server, func()) {
+	s := newServer()
+	s.srv = httptest.NewUnstartedServer(http.HandlerFunc(s.serveHTTP))
+	if err := http2.ConfigureServer(s.srv.Config, new(http2.Server)); err != nil {
+		return nil, nil
```

Comment:
Since the constructors are no-fail, I think it would be better to `panic(err)` here.  This way the context of the failure isn't lost, plus returning a `nil` `Server` will panic later anyways.

Reply:
Done in the tests